test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/DataServiceContextTrackOnlyMleNonAsyncApiTests.cs

```diff
@@ -0,0 +1,335 @@
+            Assert.Equal(0, NonTrackingContext.Entities.Count);
+            Assert.Equal(3, DefaultTrackingContext.Entities.Count);
+        }
+
+        [Fact]
+        public void TrackMediaEntitiesShouldBePopulatedWithMediaEntities()
+        {
+            SetupContextWithRequestPipeline(new DataServiceContext[] { NonTrackingContext }, true);
+
+            var mleNoTracking = NonTrackingContext.Documents.ToList();
+
+            Assert.Equal(0, NonTrackingContext.EntityTracker.Entities.Count());
+
+            // verify that the stream links are equal 
+            foreach (var document in mleNoTracking)
+            {
+                var doc = document as BaseEntityType;
+                Assert.NotNull(doc.StreamDescriptor.SelfLink);
+                Assert.NotNull(NonTrackingContext.GetReadStreamUri(document));
+
+                //try and get the content and verify that the content matches the values
+                Stream result = GetTestReadStreamResult(NonTrackingContext, document);
+
+                Assert.Equal("Hello World!", new StreamReader(result).ReadToEnd());
+            }
+        }
+
+        [Fact]
+        public void TrackMediaEntitiesShouldFailForNonMleEntitiesWithoutDescriptors()
+        {
+            SetupContextWithRequestPipeline(new DataServiceContext[] { NonTrackingContext }, false);
+            var entities = NonTrackingContext.Users.ToList();
+
+            Assert.Equal(0, NonTrackingContext.EntityTracker.Entities.Count());
+
+            // verify that the stream links are equal 
+            foreach (var entity in entities)
+            {
+                var doc = entity as BaseEntityType;
+                Assert.Null(doc.StreamDescriptor);
+                Action act1 = () => NonTrackingContext.GetReadStreamUri(entity);
+
+                act1.ShouldThrow<InvalidOperationException>(Strings.Context_EntityInNonTrackedContextLacksMediaLinks);
+                //try and get the content and verify that the content matches the values
+                Action act2 = () => GetTestReadStreamResult(NonTrackingContext, entity);
+                act2.ShouldThrow<InvalidOperationException>().WithMessage(Strings.Context_EntityInNonTrackedContextLacksMediaLinks);
+            }
+        }
+
+        private Stream GetTestReadStreamResult(DataServiceContext dataServiceContext, object entity)
+        {
+            dataServiceContext.Configurations.RequestPipeline.OnMessageCreating = args =>
+            {
+                var baseEntity = entity as BaseEntityType;
+                // if we read the wrong link then reply with gibberish
+                var resp = (args.RequestUri == baseEntity?.StreamDescriptor?.SelfLink) ? AttatchmentResponse : "gibberish";
+
+                return new CustomizedRequestMessage(
+                    args,
+                    resp,
+                    new Dictionary<string, string>
+                    {
+                            {"Content-Type", "text/plain"}

```

Comment:
Why not use the same content type as on the stream descriptor, for those that have a content type?

Reply:
Wont cause any issues on the client, the server could in theory send a different content type. But will change it to be consistent to reduce confusion.